django/forms/widgets.py

```diff
@@ -61,6 +63,57 @@ class MediaOrderConflictWarning(RuntimeWarning):
     pass
 
 
+@html_safe
+class MediaAsset:
+    element_template = "{path}"
+
+    def __init__(self, path, **attributes):
+        self._path = path
+        self.attributes = attributes
+
+    def __eq__(self, other):
+        # Compare the path only, to ensure performant comparison in Media.merge.
+        return (self.__class__ is other.__class__ and self.path == other.path) or (
+            isinstance(other, str) and self._path == other
+        )
+
+    def __hash__(self):
+        # Hash the path only, to ensure performant comparison in Media.merge.
+        return hash(self._path)
+
+    def __str__(self):
+        return format_html(
+            self.element_template,
+            path=self.path,
+            attributes=flatatt(self.attributes),
+        )
+
+    def __repr__(self):
+        return f"{type(self).__qualname__}({self._path!r})"
+
+    @property
+    def path(self):
+        """
+        Ensure an absolute path.
+        A relative paths are resolved via the `static` template tag.
+        """
+        if self._path.startswith(("http://", "https://", "/")):
+            return self._path
+        return static(self._path)
+
+
+class CSS(MediaAsset):
+    element_template = '<link href="{path}"{attributes}>'
+
+    def __init__(self, path, **attributes):
+        super().__init__(path, **attributes)
+        self.attributes["rel"] = "stylesheet"
+
+
+class JS(MediaAsset):
+    element_template = '<script src="{path}"{attributes}></script>'
```

Comment:
Do we need to add error handling for if `href`, `src`, and `rel` is passed in as attributes? Maybe also a tweak to the docs?

Reply:
Thank you for the updates :star: 
Maybe the docs need tweaking to be `.. class:: JS(src, **attributes)` and `.. class:: CSS(href, **attributes)` (with the updated param name) and maybe we add docs for the path attribute :thinking: